src/modules/auth/credentials/BulkAssignmentContainer.js

```diff
@@ -0,0 +1,460 @@
+import React, { useState, useEffect, useMemo } from "react"
+import { useTranslation } from 'react-i18next';
+import { Link } from "react-router-dom";
+import { Card, CardBody, CardHeader,
+	CardFooter, Button, ButtonGroup } from "reactstrap";
+import {DataTable, ButtonWithAuthz } from 'asab-webui';
+import RoleDropdown from "../components/RoleDropdown";
+import { useSelector } from "react-redux";
+
+const BulkAssignmentContainer = (props) => {
+
+	const { t } = useTranslation();
+
+	const [data, setData] = useState([]);
+	const [count, setCount] = useState(0);
+	const [page, setPage] = useState(1);
+	const [limit, setLimit] = useState(0);
+	const [loading, setLoading] = useState(true);
+	const [credentialsFilter, setCredentialsFilter] = useState("");
+	const [selectedCredentials, setSelectedCredentials] = useState([]);
+
+	const [tenants, setTenants] = useState({});
+	const [tenantsPage, setTenantsPage] = useState(1);
+	const [tenantsCount, setTenantsCount] = useState(0);
+	const [tenantsLimit, setTenantsLimit] = useState(0);
+	const [tenantsFilter, setTenantsFilter] = useState("");
+	const [loadingTenants, setLoadingTenants] = useState(true);
+	const [selectedTenants, setSelectedTenants] = useState([{_id: 'Global roles', selectedRole: []}]);
```

Comment:
@petrKavulok `Global roles` is not a good name, also I wouldnt mix that with tenants.

Instead, I would create a new state `[globalRoles, setGlobalRoles] = useState({global: false, selectedRoles: []});` and use that in the code whenever you are adding a global role (if global role will be added, then you will set the `global` to true and then you can also update `selectedRoles` if user will add any. When Saving the changes, you will just merge that with the request which will go to the service. At the end of the day, it will make the code way more clear. I still think that you should not mix tenants with global roles and (at least in code) it should be separated.

But ofcourse, dont hesitate to tell me, if it is not possible, or if it makes the code more unclear/harder to read, or if it will have an impact on performance...

Reply:
@petrKavulok Well, honestly, I'd rather do it in this PR. Trust me, we (you) will be crying when refactoring that in few weeks/months.